### DISCUSSION TITLE
Ploting over SSH remote connection

### DISCUSSION THREAD

**Discussion by mppalves** [REditorSupport/vscode-R#1055] [Q&A]
Hello, 

First of all, thank you for the work put into this extension. It is a life saver.
I'm working on an SSH remote connection using the "Remote - SSH" extension. I have installed the R extension in the remote server and everything works fine except for the plots. 
Does anybody have the same issue or a solution?
Thanks!

**Comment by renkun-ken** [comment]
Are you using httpgd and enable `r.plot.useHttpgd`?

**Reply by mppalves** [reply]
yes

![image](https://user-images.githubusercontent.com/37006832/160853594-a474c405-182e-4347-b617-f36b0744ffed.png)

**Comment by renkun-ken** [comment]
Is your R session attached by the session watcher when you create an R terminal?

The R process id is shown in the status bar if it is attached. You could run `ls.str(.vsc)` and see if the session watcher works normally.

**Reply by mppalves** [reply]
That seems to be the problem, my session is not attaching properly. My status bar says: R:(not attached).
Any advice on how to fix that?
Thanks

**Reply by mppalves** [reply]
Ok, I managed to attach the session, so the Viewer works now. But the plots still don't work.
![image](https://user-images.githubusercontent.com/37006832/160862748-8859796e-ccce-41b9-adb6-abd5f3abaf87.png)

**Reply by renkun-ken** [reply]
In this attached session, would you like to put `ls.str(.vsc)` here?

**Comment by mppalves** [comment]
```
addin_registry :  chr "/p/tmp/R_tmp/RtmpDavgVY/vscode-R/addins.json"
address : function (x)  
attach : function ()  
capture_str : function (object, max.level = getOption("vsc.str.max.level", 0))  
dataview_table : function (data)  
dir_session :  chr "/p/tmp/R_tmp/RtmpDavgVY/vscode-R"
dir_watcher :  chr "/home/pedrosa/.vscode-R"
get_column_def : function (name, field, value)  
get_response_lock : function ()  
get_response_timestamp : function ()  
get_timestamp : function ()  
globalenv_cache : <environment: 0x560e3291e890> 
homedir :  chr "/home/pedrosa"
inspect_env : function (env, cache)  
load_settings : function ()  
path_to_uri : function (path)  
pid :  int 28573
print.help_files_with_topic : function (h, ...)  
print.hsearch : function (x, ...)  
rebind : function (sym, value, ns)  
request : function (command, ...)  
request_browser : function (url, title, ..., viewer)  
request_file :  chr "/home/pedrosa/.vscode-R/request.log"
request_lock_file :  chr "/home/pedrosa/.vscode-R/request.lock"
request_response : function (command, ...)  
response_file :  chr "/p/tmp/R_tmp/RtmpDavgVY/vscode-R/response.log"
response_lock_file :  chr "/p/tmp/R_tmp/RtmpDavgVY/vscode-R/response.lock"
response_time_stamp :  chr ""
response_timeout :  num 5
rstudioapi_enabled : function ()  
rstudioapi_env : <environment: 0x560e32a22438> 
rstudioapi_util_env : <environment: 0x560e32a22748> 
scalar : function (x)  
settings_file :  chr "/home/pedrosa/.vscode-R/settings.json"
show_browser : function (url, title = url, ..., viewer = getOption("vsc.browser", "Active"))  
show_dataview : function (x, title, uuid = NULL, viewer = getOption("vsc.view", "Two"), row_limit = abs(getOption("vsc.row_limit", 0)))  
show_globalenv :  logi TRUE
show_page_viewer : function (url, title = NULL, ..., viewer = getOption("vsc.page_viewer", "Active"))  
show_plot :  logi FALSE
show_view :  logi TRUE
show_viewer : function (url, title = NULL, ..., viewer = getOption("vsc.viewer", "Two"))  
show_webview : function (url, title, ..., viewer)  
tempdir :  chr "/p/tmp/R_tmp/RtmpDavgVY"
try_capture_str : function (object, max.level = getOption("vsc.str.max.level", 0))  
try_catch_timeout : function (expr, timeout = Inf, ...)  
update_workspace : function (...)  
use_httpgd :  logi FALSE
user_options :  chr [1:69] "add.smooth" "bitmapType" "browser" "browserNLdisabled" "callr.condition_handler_cli_message" "CBoundsCheck" "check.bounds" "citation.bibtex.max" "continue" ...
wd :  chr "/p/projects/landuse/users/pedrosa"
workspace_file :  chr "/p/tmp/R_tmp/RtmpDavgVY/vscode-R/workspace.json"
workspace_lock_file :  chr "/p/tmp/R_tmp/RtmpDavgVY/vscode-R/workspace.lock
```

**Reply by renkun-ken** [reply]
In your `.vsc`,

```
use_httpgd :  logi FALSE
```

looks like `r.plot.useHttpgd: true` does not take effect in your R session for some reason.

Would you like to try appending the following to your `~/.Rprofile` and see if the httpgd viewer works properly?

```r
options(vsc.use_httpgd = TRUE)
```

**Reply by mppalves** [reply]
I just found out I'm having some trouble installing httpgd in my cluster.  The R version I'm using requires some packaged update to install httpgd that I cannot make without administrator rights. I will have to ask my administrator to install it for me. 
In the meanwhile, I managed to plot using x11, which solves my problem at the moment! 
Thanks for the support!!
